pipelines/controller/src/main/java/org/openmrs/analytics/DwhFilesManager.java

```diff
@@ -335,4 +356,39 @@ public Set<ResourceId> getAllChildDirectories(String baseDir) throws IOException
     logger.info("Child directories : {}", childDirectories);
     return childDirectories;
   }
+
+  private int getLastIndexOfSlash(String dwhRootPrefix) {
+    String scheme = parseScheme(dwhRootPrefix);
+    int index = -1;
+    switch (scheme) {
+      case DEFAULT_SCHEME:
+        index = dwhRootPrefix.lastIndexOf("/");
+        break;
+      case GcsPath.SCHEME:
+        String gcsObject = GcsPath.fromUri(dwhRootPrefix).getObject();
+        int position = gcsObject.lastIndexOf("/");
+        if (position != -1) index = dwhRootPrefix.indexOf(gcsObject) + position;
```

Comment:
Just checking: what happens if the `dwhRootPrefix` is `gs://my_bucket/my_object_prefix_`? In this case, does `GcsPath.fromUri(dwhRootPrefix).getObject()` returns `/my_object_prefix` or just `my_object_prefix`? If the latter, then doesn't that lead to a bug (index being `-1`)?

Reply:
If `dwhRootPrefix` is `gs://my_bucket/my_object_prefix_` then `GcsPath.fromUri(dwhRootPrefix).getObject()` return `my_object_prefix`. This leads to index being -1 and throws an error. We are handling this error as we are expecting the gcs url to be of the format `gs://<bucket>/<baseDirPath>/<prefix>`, if `baseDirPath` is empty then application throws an error.